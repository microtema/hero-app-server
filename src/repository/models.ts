import readdirSync from 'recursive-readdir-synchronous';
import Model, {Sequelize} from 'sequelize';

const {DATABASE, DATABASE_USER, DATABASE_PWD} = process.env;

const sequelize = new Sequelize(DATABASE, DATABASE_USER, DATABASE_PWD,
    {
        dialect: 'postgres',
    },
);

console.log('Connected to database successfully');

const filterModel = (it) => it.match('.orm.ts');
const cleanupModelPath = (it) => '..' + it.substring(it.indexOf('/src') + 4).replace('.ts', '');
const importModel = (it) => sequelize.import(it);
const capitalizeFirstLetter = (str) => str.charAt(0).toUpperCase() + str.slice(1);

// Note: Autowired models
const models = readdirSync('./src')
    .filter(filterModel)
    .map(cleanupModelPath)
    .map(importModel)
    .reduce((map, it) => {
        const key = capitalizeFirstLetter(it.name + 'Model');
        map[key] = it;
        return map;
    }, {});

const associate = (modelName) => {
    models[modelName].associate(models);
};

Object.keys(models).forEach(associate);

export {sequelize};

export default models;
