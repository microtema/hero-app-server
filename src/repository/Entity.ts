export interface Entity<ID> {
    id: ID;
}
