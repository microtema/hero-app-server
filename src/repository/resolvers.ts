import {merge} from 'lodash';
import {loadModules} from '../util/File.util';

export default loadModules('.resolver.ts').reduce((map, it) => merge(map, it), {});
