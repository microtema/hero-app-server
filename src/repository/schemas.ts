import {gql} from 'apollo-server-express';
import {loadModules} from '../util/File.util';

const linkSchema = gql`
    type Query {
        _: Boolean
    }
    type Mutation {
        _: Boolean
    }
    type Subscription {
        _: Boolean
    }
`;

export default [linkSchema, ...loadModules('.schema.ts')];
