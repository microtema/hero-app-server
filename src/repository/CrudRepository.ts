import {DestroyOptions, FindOptions, Identifier, UpdateOptions} from 'sequelize';
import {Entity, EntityStatic} from './Entity';

abstract class CrudRepository<T extends Entity<ID>, ID extends Identifier> {

    /**
     * Find all Authors filtered by name
     *
     * @param options may be undefined or null
     */
    public findAll(options?: any) {

        return this.model().findAll(options || {});
    }

    /**
     * Find entity by id
     *
     * @param id may not be null
     * @param options may be null
     */
    public findByPk(id: ID, options?: any) {

        options = options || {
            where: {
                id,
            },
        } as FindOptions;

        return this.model().findByPk(id, options);
    }

    /**
     * Save new entity
     *
     * @param entity may not be null
     */
    public save(entity: T) {

        return this.model().create(entity);
    }

    /**
     * Update new Author
     *
     * @param entity may not be null
     */
    public update(entity: T) {

        const predicate = {
            where: {
                id: entity.id,
            },
        } as UpdateOptions;

        return this.model().update(entity, predicate);
    }

    /**
     * Delete enity by Id
     *
     * @param id may not be null
     */
    public delete(id: ID) {

        return this.model().destroy({where: {id}} as DestroyOptions);
    }

    public abstract model(): EntityStatic<ID>;
}

export default CrudRepository;
