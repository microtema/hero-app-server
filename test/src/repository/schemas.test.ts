import {default as sut} from '../../../src/repository/schemas';

describe('Schemas', () => {

    it('Should be defined', () => {
        expect(sut).toBeDefined();
    });

    it('Should match', () => {
        expect(sut).toMatchSnapshot();
    });
});
