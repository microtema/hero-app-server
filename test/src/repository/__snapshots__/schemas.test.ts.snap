// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schemas Should match 1`] = `
Array [
  Object {
    "definitions": Array [
      Object {
        "description": undefined,
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "_",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Boolean",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "Query",
        },
      },
      Object {
        "description": undefined,
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "_",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Boolean",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "Mutation",
        },
      },
      Object {
        "description": undefined,
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "_",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Boolean",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "Subscription",
        },
      },
    ],
    "kind": "Document",
    "loc": Object {
      "end": 137,
      "start": 0,
    },
  },
  Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "address",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Address",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "street",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "addresses",
            },
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Address",
                },
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Query",
        },
      },
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "street",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "zip",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "city",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "country",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "authorId",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "createAddress",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Address",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "name",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "updateAddress",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Address",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "deleteAddress",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Boolean",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Mutation",
        },
      },
      Object {
        "description": undefined,
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "id",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "ID",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "street",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "zip",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "city",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "country",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "author",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Author",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "Address",
        },
      },
    ],
    "kind": "Document",
    "loc": Object {
      "end": 488,
      "start": 0,
    },
  },
  Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "author",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Author",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "name",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "authors",
            },
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Author",
                },
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Query",
        },
      },
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "name",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "createAuthor",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Author",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "name",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "updateAuthor",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Author",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "deleteAuthor",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Boolean",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Mutation",
        },
      },
      Object {
        "description": undefined,
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "id",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "ID",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "name",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "books",
            },
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Book",
                },
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "addresses",
            },
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Address",
                },
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "Author",
        },
      },
    ],
    "kind": "Document",
    "loc": Object {
      "end": 379,
      "start": 0,
    },
  },
  Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "book",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Book",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "title",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "books",
            },
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Book",
                },
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Query",
        },
      },
      Object {
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "title",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "authorId",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "createBook",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Book",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "title",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "updateBook",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Book",
              },
            },
          },
          Object {
            "arguments": Array [
              Object {
                "defaultValue": undefined,
                "description": undefined,
                "directives": Array [],
                "kind": "InputValueDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            ],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "deleteBook",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Boolean",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeExtension",
        "name": Object {
          "kind": "Name",
          "value": "Mutation",
        },
      },
      Object {
        "description": undefined,
        "directives": Array [],
        "fields": Array [
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "id",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "ID",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "title",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          Object {
            "arguments": Array [],
            "description": undefined,
            "directives": Array [],
            "kind": "FieldDefinition",
            "name": Object {
              "kind": "Name",
              "value": "author",
            },
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Author",
              },
            },
          },
        ],
        "interfaces": Array [],
        "kind": "ObjectTypeDefinition",
        "name": Object {
          "kind": "Name",
          "value": "Book",
        },
      },
    ],
    "kind": "Document",
    "loc": Object {
      "end": 352,
      "start": 0,
    },
  },
]
`;
